pkg/kv/kvserver/mvcc_gc_queue.go

```diff
@@ -520,12 +523,26 @@ func makeMVCCGCQueueScoreImpl(
 	return r
 }
 
+// gcHintedRangeDelete returns true iff GC should be run because the entire
+// range keyspace is likely covered by range tombstones.
 func gcHintedRangeDelete(hint roachpb.GCHint, ttl time.Duration, now hlc.Timestamp) bool {
-	deleteTimestamp := hint.LatestRangeDeleteTimestamp
-	if deleteTimestamp.IsEmpty() {
-		return false
-	}
-	return deleteTimestamp.Add(ttl.Nanoseconds(), 0).Less(now)
+	return canGC(hint.LatestRangeDeleteTimestamp, ttl, now)
+}
+
+// shouldRunGC returns whether the GCHint suggests running GC.
+func shouldRunGC(hint roachpb.GCHint, ttl time.Duration, now hlc.Timestamp) bool {
```

Comment:
This name is too general, since there's many other conditions that also come into whether twe should run GC. In fact, since we only really need to check `MinDeleteTimestamp` if we keep the old range delete check as-is, then we can just inline `if canGC(MinDeleteTimestamp) { ShouldQueue: true }` and drop this function.

Reply:
Done.